UsageStatsPlugin.inc.php

```diff
@@ -587,6 +600,48 @@ function displayReaderMonographGraph($hookName, $params) {
 		return false;
 	}
 
+	/**
+	 * Add chart to preprint view page
+	 *
+	 * Hooked to `Templates::Preprint::Main`
+	 * @param $hookName string
+	 * @param $params array [
+	 *  @option Smarty
+	 *  @option string HTML output to return
+	 * ]
+	 */
+	function displayReaderPreprintGraph($hookName, $params) {
+		$smarty =& $params[1];
+		$output =& $params[2];
+
+		$context = $smarty->getTemplateVars('currentContext');
+		$pluginSettingsDao = DAORegistry::getDAO('PluginSettingsDAO');
+		$contextDisplaySettingExists = $pluginSettingsDao->settingExists($context->getId(), $this->getName(), 'displayStatistics');
+		$contextDisplaySetting = $this->getSetting($context->getId(), 'displayStatistics');
+		$siteDisplaySetting = $this->getSetting(CONTEXT_ID_NONE, 'displayStatistics');
+		if (($contextDisplaySettingExists && $contextDisplaySetting) ||
+			(!$contextDisplaySettingExists && $siteDisplaySetting)) {
+
+				$pubObject = $smarty->getTemplateVars('article');
+				assert(is_a($pubObject, 'PublishedArticle'));
```

Comment:
This class doesn't exist anymore (in OJS).

Reply:
So probaly a `Submission` then?

Reply:
(the graph in the frontend is still broken. I have to check first if it is broken also for ojs master)

Reply:
Yep, broken also in OJS master and probably OMP master as well. Will try to solve this, but I think the graph there could use an overhaul

Reply:
@asmecher got the graph working. The OJS branch was already fixed, I just had some empty graphs data loading from file cache.